server/src/main/java/com/projects/pms/services/EmployeeService.java

```diff
@@ -0,0 +1,176 @@
+package com.projects.pms.services;
+
+
+
+import com.projects.pms.dtos.*;
+import com.projects.pms.empmapper.MapperImp;
+import com.projects.pms.exception.CustomException;
+import com.projects.pms.models.Employee;
+import com.projects.pms.repo.EmpRepo;
+import lombok.Setter;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.stereotype.Service;
+import java.util.*;
+
+
+@Service
+public class EmployeeService implements UserDetailsService {
+
+    @Autowired(required = true)
+    private EmpRepo empRepo;
+
+    @Autowired(required = true)
+    private MapperImp mapperImp;
+
+
+    @Lazy
+    @Autowired
+    private AuthenticationManager authenticationManager;
+
+    @Setter
+    private BCryptPasswordEncoder bcryptEncoder;
+
+
+    @Override
+    public UserDetails loadUserByUsername(String userName) throws UsernameNotFoundException {
+        Employee emp = empRepo.findByEmail(userName);
+        User user = new User(emp.getEmail(), emp.getPassword(), Collections.singleton(new SimpleGrantedAuthority(emp.getRole())));
+        return user;
+    }
+
+
+    public ResponseEntity<?> getAllEmps() {
+        List<Employee> employeeList = empRepo.findAll();
+
+        if (employeeList.isEmpty()) {
+            String errorMessage = "No employees found.";
+            CustomException errorResponse = new CustomException(errorMessage);
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).body(errorResponse);
+        }
+
+        return ResponseEntity.ok(employeeList);
+    }
+
+    public ResponseEntity<?> getProfile(String username) {
+        Employee employee = empRepo.findByEmail(username);
+        if (employee == null) {
+            String errorMessage = "Employee not found with email: " + username;
+            CustomException errorResponse = new CustomException(errorMessage);
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).body(errorResponse);
+        }
```

Comment:
@SunilSunny698 Use Optional and throw and avoid null checks on the model objects. and throw error on optional.emtpy if required ( Beautify the code revisit all the service methods )

Reply:
Okay